store/src/server/api.rs

```diff
@@ -215,18 +218,38 @@ impl api_server::Api for StoreApi {
         let nullifiers = validate_nullifiers(&request.nullifiers)?;
         let accounts = request
             .accounts
-            .into_iter()
+            .iter()
             .map(|account_update| {
                 let account_state: AccountState = account_update
                     .try_into()
                     .map_err(|err: ConversionError| Status::invalid_argument(err.to_string()))?;
-                Ok((
-                    account_state.account_id.into(),
-                    None, // TODO: Process account details (next PR)
-                    account_state
+
+                match (account_state.account_id.is_on_chain(), account_update.details.is_some()) {
+                    (true, false) => {
+                        return Err(Status::invalid_argument("On-chain account must have details"));
+                    },
+                    (false, true) => {
+                        return Err(Status::invalid_argument(
+                            "Off-chain account must not have details",
+                        ));
+                    },
+                    _ => (),
+                }
```

Comment:
I wonder if we should do a more robust check here to endure that for on-chain accounts, not only are the details provided, but also that the details are valid (i.e., applying the details to the previous state of the account gives us the current state). There are a few open questions here which I don't have the answers to - so, let's create an issue to discuss.